### PR TITLE
Implement simultaneous command use limitation

### PR DESCRIPTION
**Pull request by nozomuplus**
Simultaneous use of multiple commands accessing camera is not allowed.
Commands accessing camera are:
- ARKit
- NDI
- Image Detection
(Image Tracking and Face Tracking will be integrated to ARKit, and I didn't include them)

Updating UI is a future task.

### CODE REVIEW

File: ZIGSIMPlus/Models/AppSettingModel.swift [hunk 1]
```diff
@@ -144,11 +144,19 @@ public class AppSettingModel {
...
+            isCameraEnabled = (isCameraEnabled || isActiveByCommand[.arkit]!)
+        }
+        if command != .ndi {
+            isCameraEnabled = (isCameraEnabled || isActiveByCommand[.ndi]!)
+        }
+        if command != .imageDetection {
+            isCameraEnabled = (isCameraEnabled || isActiveByCommand[.imageDetection]!)
+        }
+        
+        return isCameraEnabled
```
Comment (fand) [inline ZIGSIMPlus/Models/AppSettingModel.swift]: It's a bit difficult to understand what this method is for..
How about renaming this method like `isCameraAvailable(for:)` ?

File: ZIGSIMPlusTests/AppSettingModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertFalse(AppSettingModel.shared.isCameraEnabled(except: .imageDetection))
+    }
+    
+    func testIsCameraEnabled_ExceptImageDetection_WhenArkitActive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = true
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertTrue(AppSettingModel.shared.isCameraEnabled(except: .imageDetection))
+    }
```
Comment (fand) [inline ZIGSIMPlusTests/AppSettingModelTests.swift]: Cool!
FYI, if you wanna cover all cases you can generate assertion like this: 

```swift
// [(arkit, ndi, imageDetection, expected)]
let cases = [
    (false, false, false, (false, false, false, false)),
    (false, false, true, (true, true, true, false)),
    (false, true, false, (true, true, false, false)),
    (true, false, false, (true, false, true, true)),
]

for case in cases {
    let aps = AppSettingModel.shared
    aps.isActiveByCommand[.arkit] = case.1
    aps.isActiveByCommand[.ndi] = case.2
    aps.isActiveByCommand[.imageDetection] = case.3

    let exp = case.4
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(), exp.1)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .arkit), exp.2)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .ndi), exp.3)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .imageDetection), exp.4)
}
```

File: ZIGSIMPlus/Models/AppSettingModel.swift (hunk 1 above)
Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: @fand This method checks if camera(s) is set to be used.
So, how about `usesCamera(exceptBy: Command)` ?
The followings are other ideas, but I think `usesCamera` is the best:
`isCameraSet`
`accessesCamera`

Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: I started thinking `isCameraSet` is better considering the combination with "exceptBy"...

Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: Oh I skipped this line.
> How about renaming this method like `isCameraAvailable(for:)` ?

I finally got your point.
This method is also used from `CommandOutputPresenter.isCameraEnabled()` (I think you created this).
I wanted to share logic with this call.

Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: I've come up with a refactoring idea.
Let me see.

Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: I've come up with a refactoring idea.
Let me see.

Comment (fand) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: Yep, this method was created by me, but the usecase has changed and the name `isCameraEnabled` doesn't fit the case anymore 😅

Comment (nozomuplus) [reply ZIGSIMPlus/Models/AppSettingModel.swift]: I renamed `isCameraEnabled` to `isCameraUsed` to clarify the meaning.
Also I defined `CommandAndServiceMediator.isCameraAvailable(for:)` to be more understandable.

File: ZIGSIMPlusTests/AppSettingModelTests.swift (hunk 2 above)
Comment (nozomuplus) [reply ZIGSIMPlusTests/AppSettingModelTests.swift]: I see, using tuples makes it simpler.
Thanks!

**Pull request merged**